eloquentjs/4_2_reverse_array.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Performs non-mutating array reversal operation
+ * @param {*} input 
+ * @returns 
+ */
+function reverseArray(input) {
+    if (!Array.isArray(input)) {
+        throw "input must be array";
+    }
+
+    output = [];
+    for (let i = 0; i < input.length; i++) {
```

Comment:
This is really, really close. the only problem is that `unshift` needs to copy every single array element forward when it's called making the performance of this function O(n^2). Can you update this to use `push` instead?